### DISCUSSION TITLE
Limon şekeri sorusu

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/BilsemPython#53] [Soru Cevap] [answered]
Ada ve Deniz bir kutudan sırayla limon şekerleri alıyor. Önce Ada 1 şeker, sonra Deniz 2 şeker, sonra Ada 3 şeker, sonra Deniz 4 şeker biçiminde, sırası gelen kişi kutuda yeterince şeker kalmışsa, bir önceki seferde diğer kişinin aldığı şekerin bir fazlasını alıyor. Kutuda yeterli sayıda şeker kalmamışsa sırası gelen kişi kutudaki bütün şekerleri alıyor. Ada sonuçta toplam 2900 şeker almışsa, başlangıçta kutuda kaç tane limon şekeri vardı?

(Önce algoritmasını kendi cümlelerinizle yazınız daha sonra kodunuzu ekleyiniz, kodunuzu kod bloğuna almayı unutmayınız)

**Comment by CinarSatir** [comment]
```python
#degiskenleri tanımladım
ada=1
a = 1
deniz=2
d = 2
#denizin ve adanın adanın 2900 uncu sekerini almadan once kac seker aldıgını buldum 
while(a<2900):
    if a+ada>2900:
        break
    ada = ada+2
    a = a+ada
    deniz = deniz+2
    d = d+deniz
print("Kutuda",d+2900,"limon sekeri vardır")
#ikisininde sekerlerini topladım
```

**Comment by akgembi12** [comment, marked as answer]
ilk önce ada ve denizi ilk hamlelerini elle tanımladım:
``` py
asıra=1
atoplam=1
dsıra=2
dtoplam=2 
```

sonra da kaç hamle yaptıklarını bilmediim için sonsuz bir `while` döngüsü açtım.
```py
while True:
```
bir seferde alacakları şeker sayısını ikişer ikişer artırdım
```py
asıra+=2
dsıra+=2
```
sonra da toplam şeker sayısına o hamlede alacakları şeker sayısını ekledim.
```py
atoplam += asıra
dtoplam += dsıra
```
fakat adanın 2900 şeker barajını geçip geçmediğini kontrol etmem lazımdı. bu yüzden `while` döngüsünün başına bir `if` bloğu ekledim.
```py
if atoplam+asıra>=2900:
    break
```
dönünün sonunda da toplam şeker sayısını yazdırdım.
```py
print(dtoplam+2900)
```
kodlar toplu halde şu şekilde oldu:
```py
asıra=1
atoplam=1
dsıra=2
dtoplam=2

while True:
    if atoplam+asıra>=2900:
        break
    asıra+=2
    dsıra+=2

    atoplam += asıra
    dtoplam += dsıra


print(dtoplam+2900)

```

verdiği çıktı ve toplam şeker sayısı 5762

**Comment by OmerMete** [comment]
Burda ilk değişkenleri hazırladım. While ın içinde deniz ile adanın 2900. şekeri almadan önceki şeker sayılarını buldum. Sondada denizin şeker sayısını ve zaten adanın sahip olduğu 2900 şekeri toplayı yazdırdım.
```
ada=1
a=1
deniz=2
d=2
while a<2900:
  if a+ada>2900:
    break
  ada=ada+2
  a=a+ada
  deniz=deniz+2
  d=d+deniz
print(d+2900,"tane limon şekeri kutunun içindedir")
```

**Comment by Berkek007** [comment]
```python
#Ada ve Deniz'in ilk aldığı değerleri girdim ve gerekenleri yedekledim
ada = 1
ada2 = 1
deniz = 2
deniz2 = 2

#İki arkadaşın, Ada 2900 sınırına ulaşana kadarki şeker sayılarını hesaplamak için bir döngü kurdum
while(ada2<2900):
    if ada2+ada>2900:
        break
    ada+=2
    ada2+=ada
    deniz+=2
    deniz2+=deniz
print("Kutuda toplam: ",deniz2+2900,"tane limon sekeri vardır!")
#Ada'nın 2900 şekerine Deniz'i ekledim
```
